consent/models.py

```diff
@@ -65,10 +65,21 @@ class ProxyDataSharingConsent(ConsentModelMixin):
 
     objects = DataSharingConsentManager()
 
-    def __init__(self, enterprise_customer=None, username='', course_id='', granted=False):
+    def __init__(self, enterprise_customer=None, username='', course_id='', granted=False, **kwargs):
         """
         Initialize a proxy version of ``DataSharingConsent`` which behaves similarly but does not exist in the DB.
         """
+        ec_keys = {}
+        for key in kwargs:
+            if str(key).startswith('enterprise_customer__'):
+                ec_keys.update({key[len('enterprise_customer__'):]: kwargs[key]})
+
+        if ec_keys:
+            try:
```

Comment:
Yeah, I think the standard behavior is to not catch the `DoesNotExist` exception and let the user handle it, so let's not catch it then.